diff --git a/app/src/main/java/me/magnum/melonds/ui/emulator/EmulatorActivity.kt b/app/src/main/java/me/magnum/melonds/ui/emulator/EmulatorActivity.kt
index f2696f2..437945c 100644
--- a/app/src/main/java/me/magnum/melonds/ui/emulator/EmulatorActivity.kt
+++ b/app/src/main/java/me/magnum/melonds/ui/emulator/EmulatorActivity.kt
@@ -33,6 +33,7 @@ import androidx.core.view.isGone
 import androidx.core.view.isInvisible
 import androidx.core.view.isVisible
 import androidx.core.view.updateLayoutParams
+import androidx.core.view.doOnPreDraw
 import androidx.lifecycle.DEFAULT_ARGS_KEY
 import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.lifecycleScope
@@ -777,7 +778,7 @@ class EmulatorActivity : AppCompatActivity() {
                 setLayoutComponentToggleState(LayoutComponent.BUTTON_FAST_FORWARD_TOGGLE, frontendInputHandler.fastForwardEnabled)
                 setLayoutComponentToggleState(LayoutComponent.BUTTON_MICROPHONE_TOGGLE, frontendInputHandler.microphoneEnabled)
             }
-            handler.post {
+            binding.viewLayoutControls.doOnPreDraw {
                 updateRendererScreenAreas()
             }
 
@@ -1020,4 +1021,4 @@ class EmulatorActivity : AppCompatActivity() {
         frameRenderCoordinator.stop()
         presentation?.dismiss()
     }
-}
\ No newline at end of file
+}
